src/Data/HashMap/Linear.hs

```diff
@@ -216,62 +216,65 @@ insertAll ((k, v) : xs) hmap = insertAll xs (insert hmap k v)
 -- a full hashmap, return a probe result: the place the key already
 -- exists, a place to swap from, or an unfilled cell to write over.
 probeFrom :: Keyed k =>
-  (k, PSL) -> Int -> HashMap k v #-> (HashMap k v, ProbeResult k)
-probeFrom (k, p) ix (HashMap sz@(Size s) ct arr) = read arr ix & \case
-  (arr', Ur (PSL (-1), _)) ->
-    (HashMap sz ct arr', IndexToInsert (k, p) ix)
-  (arr', Ur (psl,(k',_))) -> case k == k' of
-    -- Note: in the True case, we must have p == psl
-    True -> (HashMap sz ct arr', IndexToUpdate (k, psl) ix)
-    False -> case psl < p of
-      True -> (HashMap sz ct arr', IndexToSwap (k, p) ix)
-      False -> probeFrom (k, p+1) ((ix+1)`mod` s) (HashMap sz ct arr')
-
+  (k, PSL) -> Int -> HashMap k v #-> (HashMap k v, ProbeResult k v)
+probeFrom (k, p) ix (HashMap ct arr) = Array.read arr ix & \case
+  (arr', Ur Nothing) ->
+    (HashMap ct arr', IndexToInsert p ix)
```

Comment:
Same as below. Even if obvious, we should document that the `p` is the PSL of the query performed of the empty cell.